test/v2/controllers/RetrieveForeignPropertyAdjustmentsControllerSpec.scala

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.controllers
+
+import play.api.libs.json.Json
+import play.api.mvc.Result
+import uk.gov.hmrc.domain.Nino
+import uk.gov.hmrc.http.HeaderCarrier
+import v2.models.errors._
+import v2.fixtures.foreignProperty.RetrieveForeignPropertyAdjustmentsFixtures._
+import v2.models.outcomes.ResponseWrapper
+import v2.mocks.hateoas.MockHateoasFactory
+import v2.mocks.requestParsers.MockRetrieveAdjustmentsRequestParser
+import v2.mocks.services._
+import v2.models.audit.{AuditError, AuditEvent, AuditResponse, GenericAuditDetail}
+import v2.models.hateoas.{HateoasWrapper, Link}
+import v2.models.hateoas.Method.GET
+import v2.models.request.{RetrieveAdjustmentsRawData, RetrieveAdjustmentsRequestData}
+import v2.models.response.retrieveBsasAdjustments.foreignProperty.{RetrieveForeignPropertyAdjustmentsHateoasData, RetrieveForeignPropertyAdjustmentsResponse}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class RetrieveForeignPropertyAdjustmentsControllerSpec extends ControllerBaseSpec
+  with MockEnrolmentsAuthService
+  with MockMtdIdLookupService
+  with MockRetrieveAdjustmentsRequestParser
+  with MockRetrieveForeignPropertyAdjustmentsService
+  with MockHateoasFactory
+  with MockAuditService  {
+
+  trait Test {
+    val hc = HeaderCarrier()
+
+    val controller = new RetrieveForeignPropertyAdjustmentsController(
+      authService = mockEnrolmentsAuthService,
+      lookupService = mockMtdIdLookupService,
+      requestParser = mockRequestParser,
+      service = mockService,
+      hateoasFactory = mockHateoasFactory,
+      auditService = mockAuditService,
+      cc = cc
+    )
+
+    MockedMtdIdLookupService.lookup(nino).returns(Future.successful((Right("test-mtd-id"))))
+    MockedEnrolmentsAuthService.authoriseUser()
+  }
+
+  private val nino = "AA123456A"
+  private val correlationId = "X-123"
+  private val bsasId = "717f3a7a-db8e-11e9-8a34-2a2ae2dbcce5"
+
+  private val request = RetrieveAdjustmentsRequestData(Nino(nino), bsasId)
+  private val requestRawData = RetrieveAdjustmentsRawData(nino, bsasId)
+
+  val testHateoasLinkRetrieveBsas = Link(href = s"/individuals/self-assessment/adjustable-summary/$nino/foreign-property/$bsasId?adjustedStatus=true",
```

Comment:
shouldnt have adjusted status query param

Reply:
We don't have query parameters in HATEOAS links IIRC.

Reply:
I'm pretty sure only the submit endpoints have HATEOAS links in one of their links